composer/optim/scheduler.py

```diff
@@ -521,26 +521,63 @@ def __call__(self, state: State, ssr: float = 1.0):
         t_max = _convert_time(self.t_max, state, ssr=ssr)
         current_time = state.timestamp.get(t_max.unit)
         frac_of_total = (current_time / t_max).value
+        frac_of_total = min(1.0, frac_of_total)
 
         coeff = (1 - frac_of_total)**self.power
         current_factor = self.alpha_f + coeff * (1.0 - self.alpha_f)
         return current_factor
 
 
-def _raise_if_warmup_and_max_duration_incompatible(t_warmup: Union[str, Time], t_max: Union[str, Time]):
+def _raise_if_max_duration_exceeds_t_max(t_max: Union[str, Time], state: State):
+    assert state.max_duration is not None, 'max_duration should be set whenever schedulers are invoked'
+    max_dur = state.max_duration
+    if isinstance(t_max, str):
+        t_max = Time.from_timestring(t_max)
+    if isinstance(max_dur, str):
+        max_dur = Time.from_timestring(max_dur)
+
+    max_dur_exceeds_t_max = False
+    if t_max.unit == max_dur.unit and t_max.value < max_dur.value:
+        max_dur_exceeds_t_max = True
+    elif (
+        t_max.unit == TimeUnit.BATCH and max_dur.unit == TimeUnit.EPOCH and state.dataloader_len is not None and
+        t_max.value < max_dur.value * int(state.dataloader_len)
+    ):
+        max_dur_exceeds_t_max = True
+    elif (
+        t_max.unit == TimeUnit.EPOCH and max_dur.unit == TimeUnit.BATCH and state.dataloader_len is not None and
+        t_max.value * int(state.dataloader_len) < max_dur.value
+    ):
+        max_dur_exceeds_t_max = True
+    elif t_max.unit != max_dur.unit:
+        log.info(
```

Comment:
why not warning here? Also, i think the message could be more explicit. Instead of "make sure that your LR schedule is defined at all points", we probably want to say something that tells the user that max_duration should never exceed t_max, or their LR schedule will be invalid.

Reply:
Every optimizer step will print to the logs and there wouldn't be an easy way to filter it out.

"max_duration should never exceed t_max" Some schedulers have a defined schedule for duration above t_max

Reply:
wait really? could you link which ones -- pretty sure that every scheduler that takes in a `t_max` is only supposed to be defined up to that

There are other schedulers with just warmup that don't take t_max, yeah

Reply:
Linked some examples below